### PR TITLE
Add the --update option to "storage cp".

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
`neuro storage cp --update` should copy only when the source file is newer than the destination file or when the destination file is missing.

### CODE REVIEW

**Comment by serhiy-storchaka** [comment]
Issues:

* We cannot just compare the modification time, because local and remote modification time can be out of sync. Needs to estimate the margin and compare with a threshold.

* Currently uploading with `--update` is slower than without it, because getting the remote directory listing is expensive. We should get a listing only if the remote directory is newer than the local one (its stat is already known).

File: neuromation/api/storage.py [hunk 1]
```diff
@@ -46,6 +51,7 @@
 
 MAX_OPEN_FILES = 100
 READ_SIZE = 2 ** 20  # 1 MiB
+TIME_THRESHOLD = 1.0
```
[the remark this answers is not in the document]
Comment (serhiy-storchaka) [reply neuromation/api/storage.py]: Yes, 1 sec. Tests set it even to 0 sec.

File: neuromation/api/storage.py [hunk 2]
```diff
@@ -80,17 +86,50 @@ def __init__(self, core: _Core, config: _Config) -> None:
...
+        self._min_time_diff = 0.0
+        self._max_time_diff = 0.0
 
     def _uri_to_path(self, uri: URL) -> str:
         uri = normalize_storage_path_uri(uri, self._config.auth_token.username)
         prefix = uri.host + "/" if uri.host else ""
         return prefix + uri.path.lstrip("/")
 
+    def _set_time_diff(self, request_time: float, resp: aiohttp.ClientResponse) -> None:
+        response_time = time.time()
```
[the remark this answers is not in the document]
Comment (serhiy-storchaka) [reply neuromation/api/storage.py]: What timer is used for file's `st_mtime`? I guess we should use the same timer.

[the remark this answers is not in the document]
Comment (serhiy-storchaka) [reply neuromation/api/storage.py]:
```
>>> import os; open('xxx', 'wb').close(); os.stat('xxx').st_mtime; time.monotonic(), time.time()
1567342134.609237
(1926164.913829162, 1567342134.6115942)
```
`time.monotonic()` returns a value unrelated to the time used by file system. It cannot be used here.

[the remark this answers is not in the document]
Comment (asvetlov) [reply neuromation/api/storage.py]: yes, you are right

**Pull request merged**